bootstrap/pkg/kfapp/ksonnet/ksonnet.go

```diff
@@ -80,24 +78,24 @@ func GetKfApp(kfdef *kfdefs.KfDef) kftypes.KfApp {
 	if goPathVar != "" {
 		_kfapp.Spec.Repo = re.ReplaceAllString(_kfapp.Spec.Repo, goPathVar+`$2`)
 	}
-	// build restConfig and apiConfig using $HOME/.kube/config if the file exist
-	_kfapp.restConfig = kftypes.GetConfig()
-	_kfapp.apiConfig = kftypes.GetKubeConfig()
 	return _kfapp
 }
 
 // Apply applies the ksonnet components to target k8s cluster.
 // Remind: Need to be thread-safe: this entry is share among kfctl and deploy app
 func (ksApp *ksApp) Apply(resources kftypes.ResourceEnum) error {
-	if ksApp.restConfig == nil || ksApp.apiConfig == nil {
+	// build restConfig and apiConfig using $HOME/.kube/config if the file exist
+	restConfig := kftypes.GetConfig()
```

Comment:
We cannot have apply() calling GetConfig():
When apply() is used by deploy app, there's no kube config

Reply:
Can we update here as well?